Mutex/functions/Test-Mutex.ps1

```diff
@@ -0,0 +1,54 @@
+﻿function Test-Mutex {
+    <#
+    .SYNOPSIS
+        Check if a mutex has been already used in another process.
+
+    .DESCRIPTION
+        Check if a mutex has been already used in another process.
+        This method can only check if a mutex by agiven name has been initialized in another process.
+        It does not check if the mutex is currently been locked.
+
+    .PARAMETER Name
+        Name of the mutex to acquire a lock on. If the mutex should be globally valid on a multi user system
+        prefix the name with "Global\". Otherwise it is assumed that it belongs only to the current
+        user session (equals the prefix "Local\").
+
+    .PARAMETER CheckIfLocked
+        If used Test-Mutex will also check if an existing Mutex has been already locked.
+
+    .EXAMPLE
+        PS C:\> Test-Mutex -Name MyModule.LogFile
+
+        Checks if the mutex 'MyModule.LogFile' has been created in another process.
+
+    .EXAMPLE
+        PS C:\> Test-Mutex -Name MyModule.LogFile -CheckIfLocked
+
+        Checks if the mutex 'MyModule.LogFile' has been created in another process and if it's locked.
+        Returns $false if it does not exist or if it's currently not locked.
+        Therefor: $false means you can use it/the corresponding ressource.
+    #>
+    [CmdletBinding()]
+    [OutputType([bool])]
+    Param (
+        [Parameter(Mandatory = $true, ValueFromPipeline = $true, ValueFromPipelineByPropertyName = $true)]
+        [string]
+        $Name,
+        [Parameter(Mandatory = $false, ValueFromPipeline = $false, ValueFromPipelineByPropertyName = $false)]
```

Comment:
This `[Parameter()]` attribute is redundant, as all these values default to $false anyway.
Including them means you actually need to start reading the details in the param block, to understand how each parameter behaves, rather than being able to skim this.
Would you mind dropping them?

Reply:
Dropped, next push will include it.